Source/WebCore/html/HTMLElement.cpp

```diff
@@ -1284,6 +1284,26 @@ static HTMLElement* topmostPopoverAncestor(Element& newPopover)
     return topmostAncestor.get();
 }
 
+void HTMLElement::checkAndPossiblyClosePopoverStack()
+{
+    if (!document().settings().popoverAttributeEnabled() || !document().hasTopLayerElement())
```

Comment:
Can this check be inlined in the header to avoid a function call whenever the setting is disabled or the document has no top layer?